src/App.tsx

```diff
@@ -14,48 +15,109 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((first, second) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return first.localeCompare(second);
+
+      case SortType.LENGTH:
+        return first.length - second.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App: FC = () => {
+  const [isReversed, setReverse] = useState(false);
+  const [sortType, setType] = useState(SortType.NONE);
+
+  const handleAlphabetSort = () => setType(SortType.ALPHABET);
+  const handleLengthSort = () => setType(SortType.LENGTH);
+  const handleReverse = () => setReverse(!isReversed);
+  const handleReset = () => {
+    setType(SortType.NONE);
+    setReverse(false);
+  };
+
+  const reorderedGoods = getReorderedGoods(
+    goodsFromServer,
+    { sortType, isReversed },
+  );
 
-export const App: React.FC = () => {
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames('button is-info', {
+            'is-light': sortType !== SortType.ALPHABET,
+          })}
+          onClick={handleAlphabetSort}
```

Comment:
Here it was possible to enter at once a function with the transferred necessary data. Then you wouldn't have to write so many handlers